proposals/infra/INF-0004-validator-hashing.md

```diff
@@ -251,10 +235,8 @@ Flag | Definition
 `D3D12_BYTECODE_FLAG_VALIDATION_DISABLED` | Never invoke bytecode validation.
 `D3D12_BYTECODE_FLAG_VALIDATE_WHEN_HASH_BYPASSED` | Only validate bytecode that has the `BYPASS` or `PREVIEW_BYPASS` hash.
 `D3D12_BYTECODE_FLAG_VALIDATE_ALL_BYTECODE` | Validate all bytecode regardless of hash.  Forcing validation this way could be useful if there are shaders that are hashed (implying validation ran on them), but there is an updated validator that might catch issues were been missed.  Or the app might want validation on shaders that might have been hashed manually without the compiler's validator.
-`D3D12_BYTECODE_FLAG_VALIDATE_ONLY_IF_DEBUG_LAYER_ENABLED_NO_FAILING` | Only the debug layer performs bytecode validation.  If errors are found, it reports debug messages without failing the shader.  Accordingly if the debug layer isn't enabled no validation is done. Without this flag, the runtime validates bytecode and errors produce failure (e.g. shader creation failure), and if the debug layer is enabled it will also do the same validation so it can print a corresponding error message.
-`D3D12_BYTECODE_FLAG_SKIP_RUNTIME_VALIDATION_IF_VALIDATOR_NOT_AVAILABLE` | If the runtime can't find a validator implementation then skip validation without failing.  If this flag is not set, the runtime will fail if it needs to validate bytecode and can't find a validator implementation.  If the debug layer is enabled, a validator implementation is always available from there even if `dxil.dll` or `dxcompiler.dll` aren't present.  Thus this flag, whose effect is limited to runtime validation, is mutually exclusive with the flag `D3D12_BYTECODE_FLAG_VALIDATE_ONLY_IF_DEBUG_LAYER_ENABLED_NO_FAILING` which limits validation to the debug layer only.
-`D3D12_BYTECODE_FLAG_PREFER_EXTERNAL_VALIDATOR_IMPLEMENTATION` | With this flag the order the debug layer and runtime search for validator implementations is: `dxil.dll`, `dxcompiler.dll`, finally `d3d12sdklayers.dll` (if the debug layer is enabled).  In the absence of this flag (default), the validator implementation search is this order: `d3d12sdklayers.dll` (if the debug layer is present), `dxil.dll`, `dxcompiler.dll`.  Notice that if the debug layer isn't enabled, this flag has no effect on runtime validation.
-`D3D12_BYTECODE_FLAGS_DEFAULT` | By default validation is only done on bytecode with the `BYPASS` or `PREVIEW_BYPASS` hash, only from the debug layer (so no failure on error).  Equivalent to the flags: `D3D12_BYTECODE_FLAG_VALIDATE_WHEN_HASH_BYPASSED` and `D3D12_BYTECODE_FLAG_VALIDATE_ONLY_IF_DEBUG_LAYER_ENABLED_NO_FAILING` |
+`D3D12_BYTECODE_FLAG_PREFER_EXTERNAL_VALIDATOR_IMPLEMENTATION` | With this flag the order the debug layer search for validator implementations is: `dxil.dll`, `dxcompiler.dll`, finally it's own implementation in `d3d12sdklayers.dll`.  In the absence of this flag (default), the debug layer just uses its own implementation.
```

Comment:
This makes me uneasy.  Anything involving searching for executable code and blindly executing it seems like it is ripe for exploitation.

I guess if it is only enabled for developer machines, or if the debug layer has been loaded, then it's less of an issue.

I don't understand the motivation for this feature at all.